### ISSUE TITLE
Bug report, need more checks to see the deploymentconfig / RC is 100% up..

### ISSUE THREAD

**Issue by cdrage** [openshift/odo#1542]
[kind/bug]

## What versions of software are you using?
- Operating System: The awesome Linux OS

- Output of `odo version`: Using PR: https://github.com/openshift/odo/pull/1541

## How did you run odo exactly?

Use `odo push` multiple times very quickly...

This results in an error as the application has yet to 100% come up.

## Actual behavior

So what's happening is that we're (most likely) not checking to see that DeploymentConfig has been properly deployed / the component is actually up (replication controller too judging from the OpenShift logs)

```sh
github.com/openshift/odo  bug-hunting-2-deploys ✔
▶ ./odo push --context ~/nodejs-ex 
 ✓  Checking component
 ✓  Checking component version
 ✓  Applying component settings to component: nodejs-sxmy
 ✓  The component nodejs-sxmy was updated successfully
 ✓  Successfully created component with name: nodejs-sxmy
 ✓  Pushing changes to component: nodejs-sxmy of type local
 ✓  Waiting for component to start
 ✓  Copying files to component
 ◒  Building component ✗  Unable to build files
+ set -eo pipefail
+ '[' '!' -z /opt/app-root/src ']'
+ '[' /tmp '!=' /opt/app-root/src ']'
+ '[' -n /opt/app-root/src-backup ']'
+ '[' '!' -d /opt/app-root/src-backup ']'
+ rsync -rlO /tmp/src/. /opt/app-root/src-backup/src/
+ b_IFS='
'
+ b_OFS=
+ OIFS='
'
+ IFS='
'
++ ls -A /tmp/src/
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/.dummy
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/.gitignore
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/.odo
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/.odoignore
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/README.md
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/foo
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/foobar
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/helm
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/openshift
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/package.json
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/server.js
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/tests
+ for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
+ rm -fr /opt/app-root/src/views
+ OIFS=
+ IFS='
'
+ '[' -n bucharestgold/centos7-s2i-nodejs ']'
+ '[' bucharestgold/centos7-s2i-nodejs == redhat-openjdk-18/openjdk18-openshift ']'
+ '[' -f /tmp/src/.s2i/bin/assemble ']'
+ '[' -n /usr/libexec/s2i ']'
+ rm -rf /opt/app-root/src/.git
+ /usr/libexec/s2i/assemble
---> Installing application source
---> Building your Node application from source
Current git config
url.https://github.com.insteadof=git@github.com:
url.https://.insteadof=ssh://
url.https://github.com.insteadof=ssh://git@github.com
---> Installing dependencies
---> Using 'npm install -s --only=production'

 ✗  Building component
 ✗  command terminated with exit code 137
```

## Expected behavior

No errors / full deployment comes up

**Comment by kadel** [comment]
wasn't this fixed in #1541?

**Comment by cdrage** [comment]
@kadel Yup. Closing this issue.
